src/main/java/wolox/bootstrap/security/AuthorizationFilter.java

```diff
@@ -0,0 +1,63 @@
+package wolox.bootstrap.security;
+
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import static wolox.bootstrap.constants.SecurityConstants.HEADER_NAME;
+import static wolox.bootstrap.constants.SecurityConstants.KEY;
+
+
+public class AuthorizationFilter extends BasicAuthenticationFilter {
+
+    public AuthorizationFilter(AuthenticationManager authManager) {
+        super(authManager);
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain chain) throws IOException, ServletException {
+        String header = request.getHeader(HEADER_NAME);
+
+        if (header == null) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        UsernamePasswordAuthenticationToken authentication = authenticate(request);
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        chain.doFilter(request, response);
```

Comment:
It would be nice if the can pass the user of the token to next element in the filter chain. That way we can use it in the controllers.